mpqp/core/instruction/gates/native_gates.py

```diff
@@ -598,6 +608,51 @@ def __repr__(self) -> str:
         2
     )
 
+    def to_matrix(self, nb_qubits: int = 0) -> npt.NDArray[np.complex64]:
+        """Constructs the matrix representation of a SWAP gate for two qubits in
+        a multi-qubit system.
```

Comment:
multi-qubit QCircuit ?

Reply:
done

Reply:
but is this mention even needed ? it feels like stating the obvious (multi qubits gates are not meant for single qubits circuits, of course...)

Reply:
yes i remove it